lib/GuestManager.php

```diff
@@ -48,23 +49,55 @@ public function __construct(IDBConnection $connection, EventDispatcherInterface
 	 */
 	public function updateName(Room $room, $sessionId, $displayName) {
 		$sessionHash = sha1($sessionId);
-		$result = $this->connection->insertIfNotExist('*PREFIX*talk_guests', [
-			'session_hash' => $sessionHash,
-			'display_name' => $displayName,
-		], ['session_hash']);
-
-		if ($result === 0) {
-			$query = $this->connection->getQueryBuilder();
-			$query->update('talk_guests')
-				->set('display_name', $query->createNamedParameter($displayName))
-				->where($query->expr()->eq('session_hash', $query->createNamedParameter($sessionHash)));
-			$query->execute();
+		$dispatchEvent = true;
+
+		try {
+			$oldName = $this->getNameBySessionHash($sessionHash);
+			$dispatchEvent = $oldName !== $displayName;
```

Comment:
Better, though I would rather write it like this to save the redundant comparison:
```php
if ($oldName !== $displayName) {
   ...
} else {
    $dispatchEvent = false;
}
```

But that's just my personal taste :wink: